src/main/java/uk/gov/hmcts/darts/event/service/impl/AsyncEventProcessor.java

```diff
@@ -0,0 +1,21 @@
+package uk.gov.hmcts.darts.event.service.impl;
+
+import lombok.AllArgsConstructor;
+import org.springframework.scheduling.annotation.Async;
+import org.springframework.stereotype.Service;
+import uk.gov.hmcts.darts.event.service.CleanupCurrentFlagEventProcessor;
+
+@Service
+@AllArgsConstructor
+public class AsyncEventProcessor {
+    private final CleanupCurrentFlagEventProcessor cleanupCurrentFlagEventProcessor;
+    private final RemoveDuplicateEventsProcessorImpl removeDuplicateEventsProcessor;
+
+
+    @Async("eventTaskExecutor")
+    public void processEvent(Integer eventId) {
```

Comment:
Do we need an integration test to ensure both processes (duplicate check and event current cleanup) are carried out in a happy path scenario?

Reply:
Integration tests are probably excessive here but certainly unit tests for this method would add value will add some